### PR TITLE
Move mtype.Type*.toDsymbol to typesem.d

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: compiler/src/dmd/expression.d [hunk 1]
```diff
@@ -189,39 +189,6 @@ extern (C++) void expandTuples(Expressions* exps, Identifiers* names = null)
     }
 }
 
-/****************************************
- * Expand alias this tuples.
- */
-TupleDeclaration isAliasThisTuple(Expression e)
```
Comment (thewilsonator) [inline compiler/src/dmd/expression.d:195]: this seems to be an unrelated refactor. I agree with it, but it should be a separate PR.

**Review by thewilsonator — COMMENTED** [review]
Apart from the additional unrelated refactoring, this looks good.

File: compiler/src/dmd/expression.d (hunk 1 above)
Comment (RazvanN7) [reply compiler/src/dmd/expression.d:195]: This function uses typesem.toDsymbol. I could make it import dmd.typesem and then make another PR to move it to expressionsem, but I think it's better to just move it now to expressionsem since expression.d does not rely anymore on semantic routines.

Please merge as is.
